front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/managed/permissions.ts

```diff
@@ -32,7 +30,23 @@ async function handler(
     });
   }
 
-  const dataSource = await getDataSource(auth, req.query.name as string);
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
+      },
+    });
+  }
+
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```

Comment:
nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)